safer-ffi-gen-macro/src/lib.rs

```diff
@@ -30,33 +36,40 @@ impl FFIType {
             .last()
             .ok_or_else(|| syn::Error::new(path.span(), "unexpected empty path"))?;
 
-        match last_segment.ident.to_string().as_str() {
+        let ffi_safe_type = match last_segment.ident.to_string().as_str() {
             "Vec" => {
                 let PathArguments::AngleBracketed(ref args) = last_segment.arguments else {
                     return Err(syn::Error::new(last_segment.arguments.span(), "invalid path arguments"));
                 };
-                Ok(FFIType::Vec(args.clone()))
+
+                Ok::<_, syn::Error>(FFISafeType::Vec(args.clone()))
             }
             "Box" => {
                 let PathArguments::AngleBracketed(ref args) = last_segment.arguments else {
                     return Err(syn::Error::new(last_segment.arguments.span(), "invalid path arguments"));
                 };
-                Ok(FFIType::Box(args.clone()))
+
+                Ok(FFISafeType::Box(args.clone()))
             }
-            "String" => Ok(FFIType::String),
+            "String" => Ok(FFISafeType::String),
             // Anything that isn't listed should be repr_c::Box
-            _ => Ok(FFIType::Opaque(last_segment.ident.clone())),
-        }
+            _ => Ok(FFISafeType::Opaque(last_segment.ident.clone())),
+        }?;
+
+        Ok(FFIType {
+            native_type: path,
+            ffi_safe_type,
+        })
     }
 }
 
-impl ToTokens for FFIType {
+impl ToTokens for FFISafeType {
     fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
         match self {
-            FFIType::Vec(args) => quote! { safer_ffi::prelude::repr_c::Vec #args },
-            FFIType::Box(args) => quote! { safer_ffi::prelude::repr_c::Box #args },
-            FFIType::String => quote! { safer_ffi::prelude::repr_c::String },
-            FFIType::Opaque(ident) => quote! { safer_ffi::prelude::repr_c::Box<#ident> },
+            FFISafeType::Vec(args) => quote! { safer_ffi::prelude::repr_c::Vec #args },
```

Comment:
Should the path start with `::safer_ffi` to make sure it picks up only external crates and not modules?

Reply:
Will resolve with issue #16